plugins/common/socket/stream.go

```diff
@@ -258,6 +260,17 @@ func (l *streamListener) listenData(onData CallbackData, onError CallbackError)
 				break
 			}
 
+			if allowed, err := isSourceAllowed(l.AllowedSources, conn.RemoteAddr()); err != nil {
+				if onError != nil {
+					onError(err)
+				}
+				_ = conn.Close()
```

Comment:
How about

```suggestion
				if err := conn.Close(); err != nil {
					onError(fmt.Errorf("closing connection from %q failed: %w", conn.RemoteAddr(), err))
				}
```

same below.